crates/tuono/tests/utils.rs

```diff
@@ -67,3 +72,112 @@ impl Drop for TempTuonoProject {
             .expect("Failed to restore the original directory.");
     }
 }
+
+pub struct MockServerWrapper {
+    server: MockServer,
+}
+
+pub enum ResponseBody {
+    Json(Value),
+    String(String),
+}
+
+impl MockServerWrapper {
+    pub async fn new() -> Self {
+        MockServerWrapper {
+            server: MockServer::start().await,
+        }
+    }
+
+    pub async fn register_mock(
+        &self,
+        method: &str,
+        path: &str,
+        params: Option<QueryParamExactMatcher>,
+        status: u16,
+        response_body: ResponseBody,
+    ) {
+        if cfg!(feature = "integration_test") {
+            env::set_var("GITHUB_API_BASE_URL", self.server.uri());
+            env::set_var("GITHUB_RAW_CONTENT_BASE_URL", self.server.uri());
+        } else {
+            panic!("Please run the test command with `cargo test --features integration_test´")
+        }
+
+        let mut mock = Mock::given(matchers::method(method)).and(matchers::path(path));
+
+        if let Some(params) = params {
+            mock = mock.and(params);
+        }
+
+        let response_template = match response_body {
+            ResponseBody::Json(body) => ResponseTemplate::new(status).set_body_json(body),
+            ResponseBody::String(body) => ResponseTemplate::new(status).set_body_string(body),
+        };
+
+        mock.respond_with(response_template)
+            .mount(&self.server)
+            .await;
+    }
+}
+#[cfg(test)]
```

Comment:
Please add a new line here

```suggestion

#[cfg(test)]
```